docs/api/ref/PrivateChannel.md

```diff
@@ -0,0 +1,144 @@
+
+* [`Channel`](Channel)
+* [`Listener`](Types#listener)
+* [`DesktopAgent.addIntentListener`](DesktopAgent#addintentlistener)
+* [`DesktopAgent.createPrivateChannel`](DesktopAgent#createPrivateChannel)
+* [`DesktopAgent.raiseIntent`](DesktopAgent#raiseintent)
+
+## Examples
+### 'Server-side' example:
+The intent app establishes and returns a PrivateChannel to the client (who is awaiting `getResult()`). When the client calls `addContextlistener()` on that channel, the intent app receives notice via the handler added with `onAddContextListener()` and knows that the client is ready to start receiving quotes.
+
+The Desktop Agent knows that a channel is being returned by inspecting the object returned from the handler (e.g. check constructor or look for private member).
+
+```javascript
+fdc3.addIntentListener("QuoteStream", async (context) => {
+	const channel: PrivateChannel = await fdc3.createPrivateChannel();
+	const symbol = context.id.ticker;
+
+	// This gets called when the remote side adds a context listener
+	const addContextListener = channel.onAddContextListener((contextType) => {
+		// broadcast price quotes as they come in from our quote feed
+		feed.onQuote(symbol, (price) => {
+			channel.broadcast({ type: "price", price});
+		});
+	});
+
+	// This gets called when the remote side calls Listener.unsubscribe()
+	const unsubscriberListener = channel.onUnsubscribe((contextType) => {
+		feed.stop(symbol);
+	});
+
+	// This gets called if the remote side closes
+	const disconnectListener = channel.onDisconnect(() => {
+		feed.stop(symbol);
+	});
+
+	return channel;
+});
+```
+
+### 'Client-side' example:
+
+```javascript
+try {
+    const resolution3 = await fdc3.raiseIntent("QuoteStream", { type: "fdc3.instrument", id : { symbol: "AAPL"}});
+    try {
+	    const result = await resolution3.getResult();
+        //check that we got a result and that its a channel
+        if (result && result.addContextListener) {
+            const listener = result.addContextListener("price", (quote) => console.log(quote));
+	         result.onDisconnect(() => {
+		          console.warn("Quote feed went down");
+	         });
+
+	         // Sometime later...
+	         listener.unsubscribe();
+        } else {
+            console.warn(`${resolution3.source} did not return a channel`);
+        }
+     } catch(channelError){
+	    console.log(`Error: ${resolution3.source} returned an error: ${channelError}`);
+     }
+} catch (resolverError) {
+    console.error(`Error: Intent was not resolved: ${resolverError}`);
+}
+```
+
+
+## Methods
+
+### `onAddContextListener`
+```ts
+onAddContextListener(handler: (contextType?: string) => void): Listener;
+```
+Adds a listener that will be called each time that the remote app invokes addContextListener on this channel.
+
+Desktop Agents MUST call this for each invokation of addContextListener on this channel, including those that occurred before this handler was registered (to prevent race conditions).
```

Comment:
```suggestion
Desktop Agents MUST call this for each invocation of addContextListener on this channel, including those that occurred before this handler was registered (to prevent race conditions).
```